pkg/kubelet/volume_host.go

```diff
@@ -183,6 +185,26 @@ func (kvh *kubeletVolumeHost) WaitForCacheSync() error {
 	return nil
 }
 
+// WaitForNodeRegistrationCompleted is a helper function that waits for node is in registrationCompleted state
+func (kvh *kubeletVolumeHost) WaitForNodeRegistrationCompleted(ctx context.Context) error {
+	err := wait.PollUntilContextCancel(ctx, time.Second, true, kvh.verifyNodeRegistrationCompleted())
+	if err != nil {
+		// In theory this is unreachable, but just in case:
+		return fmt.Errorf("faled to WaitForNodeRegistrationCompleted: %v", err)
+	}
+	return nil
+}
+
+func (kvh *kubeletVolumeHost) verifyNodeRegistrationCompleted() wait.ConditionWithContextFunc {
+	return func(_ context.Context) (done bool, err error) {
+		if kvh.kubelet.registrationCompleted {
+			return true, nil
+		}
+		klog.V(2).InfoS("Still waiting for node to complete registration")
```

Comment:
This may be spammy especially if it's every second. Maybe knock it down to V(4)? and add a log when it succeeds.